packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -297,15 +300,9 @@ export const makeVaultManager = (
       return;
     }
 
-    if (delta > 0n) {
-      // add the amount
-      totalDebt += delta;
-    } else {
-      // negate the amount so that it's a natural number, then subtract
-      const absDelta = -delta;
-      assert(!(absDelta > totalDebt), 'Negative delta greater than total debt');
-      totalDebt -= absDelta;
-    }
+    totalDebt += delta;
+    assert(totalDebt >= 0n, 'Negative delta greater than total debt');
```

Comment:
This assert needs to be before the totalDebt gets updated, I think. Hence compute a temporary, assert, then assign.

Reply:
Agree the lines commented have the bug that if the assert fails `totalDebt` will have been updated. I thought the thrown error would break more things but that's not a valid assumption; something could catch it and believe the `totalDebt` that was left.

However, the lines highlighted were removed in e71db04c294b0908b16d1752b71a5f641f55a4cd . That `amountMath` call will throw if the negative delta was greater than total debt (because the result wouldn't be a natural number, which is really what this is enforcing). `totalDebt` won't update.